src/_blog/rust-libp2p-2022.md

```diff
@@ -0,0 +1,87 @@
+Still remember the old days with the `NetworkBehaviourEventProcess` trait?
+[All gone](https://github.com/libp2p/rust-libp2p/pull/2784) in favor of the much simpler (generated) `OutEvent` mechanism.
+The `StreamMuxer` trait received numerous significant simplifications, [basically rewriting](https://github.com/libp2p/rust-libp2p/issues/2722) the trait as well as the trait implementation in yamux, mplex and now QUIC and WebRTC.
+
+Defense against denial-of-service attacks is a cornerstone of a networking library, especially in the peer-to-peer space.
+rust-libp2p saw a lot of related improvements in 2022.
+We enforce various limits (e.g. on the [number of streams](https://github.com/libp2p/rust-libp2p/pull/2697) and bytes of a request) and [prioritize local work over new incoming work from a remote](https://github.com/libp2p/rust-libp2p/pull/2627) across the many layers.
+Up next is [a patch](https://github.com/libp2p/rust-libp2p/issues/2824) enabling `NetworkBehaviour` implementations to implement their own connection management strategies.
+
+Understanding large systems is hard.
+Understanding distributed systems is even harder.
+We made understanding large libp2p networks a bit easier in 2022 introducing a metric crate for rust-libp2p exposing Prometheus metrics, e.g. the [time to establish a connection](https://github.com/libp2p/rust-libp2p/pull/3134) or the [protocols supported by peers](https://github.com/libp2p/rust-libp2p/pull/2734).
+
+Since January the rust-libp2p monorepo has a [handy CLI tool for libp2p key management](https://github.com/libp2p/rust-libp2p/pull/2453).
+
+In general, we keep up with [recent developments of the Rust language], and incorporate some of its new shiny features.
+We make use of [Cargo's weak dependencies](https://github.com/libp2p/rust-libp2p/pull/2646). 
+In November we introduced [a patch using const generics](https://github.com/libp2p/rust-libp2p/pull/2712) (in tests).
+In December we merged our [first pull request using GATs](https://github.com/libp2p/rust-libp2p/pull/3239).
+
+## Meta - Improvements to the Project
+
+libp2p is an open-community open-source project.
+What do I mean by that?
+libp2p's source code is open-source and that source code is written by an open community.
+libp2p is not developed by one or 10 people, but much rather >100 (part-time) people across many organizations.
+
+The core rust-libp2p maintainer team grew from two engineers to four, with [Elena](https://github.com/libp2p/rust-libp2p/pull/2656) and [João](https://github.com/libp2p/rust-libp2p/pull/3295) joining the team. Beyond the core maintainers, a total of 70 people contributed to rust-libp2p's `master` branch in 2022.
```

Comment:
```suggestion
The core rust-libp2p maintainer team grew from two engineers (myself and [Thomas](https://github.com/thomaseizinger)) to four, with [Elena](https://github.com/libp2p/rust-libp2p/pull/2656) and [João](https://github.com/libp2p/rust-libp2p/pull/3295) joining the team. Beyond the core maintainers, a total of 70 people contributed to rust-libp2p's `master` branch in 2022.
```
Let's give a shoutout to some of the notable people as well?

Reply:
Including Thomas and me. Thanks.

> Let's give a shoutout to some of the notable people as well?

I am a bit hesitant naming individuals. I wouldn't know where to draw the line.